### ISSUE TITLE
Avatars should not be color coded in a message and message list

### ISSUE THREAD

**Issue by jouni** [vaadin/vaadin-messages#39]
The avatars in a message list should not use color index. The color coding is mainly needed in places where the avatar is disconnected from the related UI element the user is associated with (think of the focus outline in input components).

Effectively, avatars with images should not have the colored outline and avatars with abbreviations or icons should use the default background color.

**Comment by pekam** [comment]
> The color coding is mainly needed in places where the avatar is disconnected from the related UI element the user is associated with (think of the focus outline in input components).

Isn't this the case also if you have an avatar group in the same view with a message list? I think it would be confusing if the avatar looks different in the avatar group and in the messages.

**Comment by jouni** [comment]
Yeah, there’s no need for color coding in Avatar Group in some cases as well. TaskMob is one such example, where it’s only used to indicate who are invited to collaborate on a project.

**Comment by pekam** [comment]
My point was that if you have both avatar group and message list, isn't that a case for _disconnected but related UI elements the user is associated with_, and thus they need color coding?

**Comment by jouni** [comment]
Each message has it’s own avatar, so there’s no need to color code it (even if we have an additional avatar group in the same UI). You can immediately see who wrote each individual message.

When we have UI elements that don’t show the avatar, we use the color coding to make the connection.

**Comment by pekam** [comment]
Ok, got it.

Just to make sure: Is it such an anti-pattern to use color index in messages, that we should remove the API we already have?

> there’s no need for color coding in Avatar Group in some cases either

Another problem is then that `CollaborationAvatarGroup` wasn't really designed with this in mind. By default, when color index is not defined (`colorIndex == -1`), it let's Collaboration Engine provide one. The color can be disabled with something hacky like `userInfo.setColorIndex(-2)`, but besides being a magic value, it has also the problem that you shouldn't need to mutate `UserInfo` when you want to disable colors for one component (possibly re-using the `UserInfo` with color elsewhere).

Maybe we need something like `collaborationAvatarGroup.setUserColorsEnabled(false)`.

**Comment by jouni** [comment]
> Is it such an anti-pattern to use color index in messages, that we should remove the API we already have?

Maybe not remove it. It might make sense in cases where you have a chat window open that is connected to something where multiple users collaborate on (like a text editor), so that you can show in the chat the color for active users. I think this is how Google Docs works.

> Another problem is then that CollaborationAvatarGroup wasn't really designed with this in mind.

What about the basic Avatar Group? I assume that can be used without colors. For Collaboration Avatar Group, it’s probably fine to assume that all avatars need a color, for now at least.

**Comment by pekam** [comment]
> For Collaboration Avatar Group, it’s probably fine to assume that all avatars need a color, for now at least.

We'll have the same problem with `CollaborationMessageList` vs. `MessageList` as with `CollaborationAvatarGroup` vs. `AvatarGroup`.

The non-collab component is configured with _component-specific_ items (`AvatarGroupItem`, `MessageListItem`). These items are used for configuring one specific component instance.

The collab components on the other hand use an _end-user-specific_ `UserInfo` object and manage the items automatically. The `UserInfo` shouldn't be used for configuring whether to show colors on particular UI component or not.

My point is: we need to decide also whether or not to have user colors in `CollaborationMessageList`, and it can't be configured unless we add new API such as `setUserColorsEnabled` for the collab components. Now it sounds like `CollaborationAvatarGroup` should have colors, but `CollaborationMessageList` should not. Maybe that's fine, but I'm a bit concerned about having different defaults for these two.

Sorry for going so deep into API details. :) That's our problem. The main questions for you as the designer would be whether or not `CollaborationAvatarGroup` and `CollaborationMessageList` should show user colors by default, and should it be configurable.

**Comment by jouni** [comment]
This starts to sounds like this should be configurable, as I can see some developers getting frustrated if they have to use a different API when they don’t want user colors for Avatar Group (i.e. they have to configure individual Avatar items instead of providing UserInfo objects). Can that change? Could the normal Avatar Group component also accept a list of UserInfo objects, but not show user colors?

I’m torn between the defaults, same as you are. 

In a chat (Message List) It’s usually unnecessary to show the color outline for an avatar that has an image. For avatars with an icon or abbreviation, coloring could be a nice addition (improve recognition), but not a requirement. But some application might find it useful to show user colors in a chat as well, so I wouldn’t want to limit that either.

For CollaborationAvatarGroup, the default has been there for some time now, so I wonder if we should change that? And perhaps the intended use case for that component is together with CollaborationBinder? If you forget to override the default, it perhaps isn’t a big issue, as the field highlights do show the name of the users when you hover/focus them. Showing the colors in the avatar group would just help in making a stronger connection when desired.

So, I think I’m leaning more towards having user colors disabled by default.

**Comment by jouni** [comment]
One more thought: the coloring could only be shown for users that are currently active/using the application. This way it wouldn’t necessarily need to be configurable, though I still foresee some apps wanting to disable the coloring completely in certain situations.

**Comment by jouni** [comment]
One more thought: the coloring could only be shown for users that are currently active/using the application. This way it wouldn’t necessarily need to be configurable, though I still foresee some apps wanting to disable the coloring completely in certain situations. Both of these behaviors can be observed in Google Docs, for example.
